paella-matterhorn/javascript/05_initdelegate.js

```diff
@@ -135,7 +126,52 @@ var isHarvardDceAuth = function (jsonData) {
 };
 // #DCE(naomi): end of dce auth addition
 // ------------------------------------------------------------
-
+// #DCE(karen): START, get search/series and tranform result into series/series format
+// This tranforms the series data into the expected upstream series format
+var searchSeriesToSeriesSeries = function (serie, onSuccess, onError) {
+    base.ajax.get({
+        url: '/search/series.json',
+        params: {
+            'id': serie
+        }
+    },
+    function (data, contentType, code) {
+        var jsonData = data;
+        try {
+            if (typeof (jsonData) == "string") jsonData = JSON.parse(jsonData);
+        }
+        catch (e) {
+            showLoadErrorMessage("Unable to parse series id" + "\"" + serie + "\" data: " + data);
+            if (typeof(onError)=='function') {
+		onError();
+	    }
```

Comment:
I think we might as well return here, since the JSON is already established to be unusable at this point, and we have duly notified the user and called `onError`.